systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -10,64 +10,57 @@
 % the answer).
 %
 
-if model.use_new_kinsol
-  checkDirty(model);  
-  if nargin<6
+checkDirty(model);
+
+% method signature transition
+if nargin > 3
+  options_copy = options;
+end
+warn_signature_changed = false;
+if nargin == 5
+  warn_signature_changed = true;
+  options = struct();
+  options.compute_gradients = v;
```

Comment:
OK. I guess we can call it compute_gradients, so long as it is clearly documented that jacobians are computed even when compute_gradients = false.  

On Feb 11, 2015, at 11:09 AM, Twan Koolen <notifications@github.com> wrote:

> In systems/plants/@RigidBodyManipulator/doKinematics.m:
> 
> > @@ -10,64 +10,57 @@
> >  % the answer).
> >  %
> >  
> > -if model.use_new_kinsol
> > -  checkDirty(model);  
> > -  if nargin<6
> > +checkDirty(model);
> > +
> > +% method signature transition
> > +if nargin > 3
> > +  options_copy = options;
> > +end
> > +warn_signature_changed = false;
> > +if nargin == 5
> > +  warn_signature_changed = true;
> > +  options = struct();
> > +  options.compute_gradients = v;
> Yes, I need to add documentation.
> 
> would it make sense to have multiple options for computing second derivatives, for computing geometry jac grads, etc.
> 
> I think the current options reflect the typical usage cases:
> 1. compute_gradients = false, compute_JdotV = false: position control, planning using centroidal dynamics and full kinematics
> 2. compute_gradients = false, compute_JdotV = true: model based control, QP controller (requires Jdot * v terms in Cartesian constraints, dynamics bias term (C), CMMdot * v)
> 3. compute_gradients = true: second order methods, 'Michael's stuff' is what I hear
> 
> I'm OK with compute_additional_gradients, although I still think compute_gradients is clearer since that's exactly what you're telling doKinematics to do.
> 
> dTdq does not get computed in doKinematics when compute_gradients = false. As an aside, you could compute a dTdq for a single body from information using a kinsol with compute_gradients = false with dHomogTrans based on the S (motion subspace) and qdot_to_v fields of kinsol, but dTdq for all bodies is not precomputed in doKinematics when compute_gradients = false.
> 
> —
> Reply to this email directly or view it on GitHub.
>